packages/catalog-process/src/services/readModelService.ts

```diff
@@ -225,18 +265,30 @@ export function readModelServiceBuilder(
         ? { "data.mode": { $eq: mode } }
         : {};
 
+      const delegatedFilter: ReadModelFilter<EService> = match(delegated)
+        .with(true, () => ({
+          "delegation.data.state": {
+            $in: [delegationState.active, delegationState.waitingForApproval],
+          },
+          "delegation.data.kind": delegationKind.delegatedProducer,
+        }))
+        .with(false, () => ({
+          "delegation.data.state": {
+            $nin: [delegationState.active, delegationState.waitingForApproval],
+          },
+        }))
```

Comment:
You are right, it missed the delegation kind check. The query now looks the following:

```javascript
{
  delegations: {
    $not: {
      $elemMatch: {
        "data.state": {
          $in: [
            delegationState.active,
            delegationState.waitingForApproval,
          ],
        },
        "data.kind": delegationKind.delegatedProducer,
      },
    },
  },
}
```

- If there are no delegations, it will match;
- If not, it will match only the records that have no active or waiting for approval delegation of kind producer.

I also updated a test in order to check that the consumer delegations are not being taken into account.
The commit is [here](https://github.com/pagopa/interop-be-monorepo/pull/1175/commits/4737927609e82f44e1853f49b55059a032109f01).